File: R/8kyu/8kyu-will-you-make-it.r
# Will you make it?
# Language: R
# Difficulty: 8 kyu

# Solution 1
zero_fuel <- function(distance, mpg, fuel_left) {
  
  if (distance - mpg * fuel_left <= 0){
    return(TRUE)
  }else{
    return(FALSE)
  }
  
}

# Solution 2 (more concise approach)
zero_fuel <- function(distance, mpg, fuel_left) {
  distance <= fuel_left * mpg
}


# Tests
test_that('basic tests', {
  expect_equal(zero_fuel(50, 25, 2), TRUE)
  expect_equal(zero_fuel(60, 30, 3), TRUE)
  expect_equal(zero_fuel(70, 25, 1), FALSE)
  expect_equal(zero_fuel(100, 25, 3), FALSE)
})

File: Python/8kyu/8kyu-will-you-make-it.py
# Will you make it?
# Language: Python
# Difficulty: 8 kyu

# Solution 1
def zero_fuel(distance_to_pump, mpg, fuel_left):
    return True if distance_to_pump - mpg * fuel_left <= 0 else False

# Solution 2 (more concise approach)
def zeroFuel(distance_to_pump, mpg, fuel_left):
    return distance_to_pump <= mpg * fuel_left


# Tests
import codewars_test as test

try: 
    from solution import zeroFuel as zero_fuel
except ImportError: 
    from solution import zero_fuel 

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(zero_fuel(50, 25, 2), True)
        test.assert_equals(zero_fuel(100, 50, 1), False)